src/http/cookie.cr

```diff
@@ -58,7 +58,7 @@ module HTTP
         # valid characters for cookie-name per https://tools.ietf.org/html/rfc6265#section-4.1.1
         # and https://tools.ietf.org/html/rfc2616#section-2.2
         # "!#$%&'*+-.0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ^_`abcdefghijklmnopqrstuvwxyz|~"
-        unless (0x21...0x7f).includes?(byte) && byte != 0x22 && byte != 0x28 && byte != 0x29 && byte != 0x2c && byte != 0x2f && !(0x3a..0x40).includes?(byte) && !(0x5b..0x5d).includes?(byte) && byte != 0x7b && byte != 0x7d
+        if !valid_cookie_value_byte?(byte) || byte.in?(0x3a..0x40, 0x5b..0x5d, 0x7b, 0x7d)
```

Comment:
This is wrong :slightly_frowning_face:
```crystal
byte.in?(0x3a..0x40, 0x5b..0x5d, 0x7b, 0x7d)

{0x3a..0x40, 0x5b..0x5d, 0x7b, 0x7d}.includes?(byte)

0x3a..0x40 != byte
```

https://github.com/crystal-lang/crystal/blob/879691b2e/src/object.cr#L217

Reply:
Oh yeah, good catch.

A bit more elaborate: `Object#in?` with a splat parameter just collects all the parameters as a collection and passes them to the one-param overload. So `byte.in?(0x3a..0x40, 0x5b..0x5d, 0x7b, 0x7d)` is equivalent to `byte.in?({0x3a..0x40, 0x5b..0x5d, 0x7b, 0x7d})`, not `byte.in?(0x3a..0x40) || byte.in?(0x5b..0x5d) || byte.in?(0x7b, 0x7d)`

Reply:
I suppose we could also use a spec for this.